extra_data/reader.py

```diff
@@ -550,25 +544,11 @@ def get_run_value(self, source, key):
         key: str
             Key of parameter within that device, e.g. "triggerMode".
         """
-        if not self.is_single_run:
-            raise MultiRunError
-
-        if source not in self.control_sources:
+        try:
+            source_data = self._sources_data[source]
+        except KeyError:
             raise SourceNameError(source)
-
-        # Arbitrary file - should be the same across a run
-        fa = self._sources_data[source].files[0]
-        ds = fa.file['RUN'][source].get(key.replace('.', '/'))
-        if isinstance(ds, h5py.Group):
-            # Allow for the .value suffix being omitted
-            ds = ds.get('value')
-        if not isinstance(ds, h5py.Dataset):
-            raise PropertyNameError(key, source)
-
-        val = ds[0]
-        if isinstance(val, bytes):  # bytes -> str
-            return val.decode('utf-8', 'surrogateescape')
-        return val
+        return source_data.run_value(key)
```

Comment:
```suggestion
        return self._get_source_data(source).run_value(key)
```

This can save the try/except.